lib/fog/compute/kubevirt/models/server.rb

```diff
@@ -0,0 +1,87 @@
+require 'fog/compute/models/server'
+
+module Fog
+  module Compute
+    class Kubevirt
+      class Server < Fog::Compute::Server
+        include Shared
+
+        identity :name
+
+        attribute :namespace,        :aliases => 'metadata_namespace'
+        attribute :resource_version, :aliases => 'metadata_resource_version'
+        attribute :uid,              :aliases => 'metadata_uid'
+        attribute :labels,           :aliases => 'metadata_labels'
+        attribute :owner_reference,  :aliases => 'metadata_owner_reference'
+        attribute :annotations,      :aliases => 'metadata_annotations'
+        attribute :cpu_cores,        :aliases => 'spec_cpu_cores'
+        attribute :memory,           :aliases => 'spec_memory'
+        attribute :disks,            :aliases => 'spec_disks'
+        attribute :volumes,          :aliases => 'spec_volumes'
+        attribute :status,           :aliases => 'spec_running'
+        attribute :state,            :aliases => 'phase'
+        attribute :ip_address
+        attribute :node_name
+
+        def start(options = {})
+          # Change the `running` attribute to `true` so that the virtual machine controller will take it and
+          # create the virtual machine instance.
+          vm = service.get_raw_vm(name)
+          vm = deep_merge!(vm,
+            :spec => {
+              :running => true
+            }
+          )
+          service.update_vm(vm)
+        end
+
+        def stop(options = {})
+          vm = service.get_raw_vm(name)
+          vm = deep_merge!(vm,
+            :spec => {
+              :running => false
+            }
+          )
+          service.update_vm(vm)
+        end
+
+        def destroy(options = {})
+          stop(options)
+          service.delete_vm(name, namespace)
+        end
+
+        def ready?
+          status == 'running' && state == 'Running' && !ip_address.blank?
+        end
+
+        def self.parse(object)
+          metadata = object[:metadata]
+          spec = object[:spec][:template][:spec]
+          domain = spec[:domain]
+          owner = metadata[:ownerReferences]
+          annotations = metadata[:annotations]
+          cpu = domain[:cpu]
+          server = {
+            :namespace        => metadata[:namespace],
+            :name             => metadata[:name],
+            :resource_version => metadata[:resourceVersion],
+            :uid              => metadata[:uid],
+            :labels           => metadata[:labels],
+            :memory           => domain[:resources][:requests][:memory],
```

Comment:
This is the reason why we can reuse. We merged a PR which allows vm not to have memory but here the code would fail. I think it is not good idea to maintain it in 2 places.

Reply:
now that should be reside in a single place - see VmAction